earth_enterprise/src/common/FileAccessor.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "FileAccessor.h"
+
+std::unique_ptr<AbstractFileAccessor> AbstractFileAccessor::getAccessor(const std::string &fname) {
+  return std::unique_ptr<POSIXFileAccessor>(new POSIXFileAccessor());
+}
+
+// ****************************************************************************
+// ***  POSIXFileAccessor
+// ****************************************************************************
+void POSIXFileAccessor::Open(const std::string &fname, const char *mode, int flags, mode_t createMask) {
```

Comment:
Minor: I suspect there's a way to combine `mode` and `flags`, but we can save that for another time.